explanation/performance/perf-pgo.md

```diff
@@ -0,0 +1,153 @@
+(perf-pgo)=
+
+# Profile-Guided Optimization
+
+[Profile-guided optimization](https://en.wikipedia.org/wiki/Profile-guided_optimization) (PGO) or Feedback-driven optimization (FDO) are synonyms for the same concept.  The idea is that a binary is compiled with built-in instrumentation which, when executed, will generate a profile.  This profile can then be used as an input for the compiler in a subsequent build of the same binary, and it will serve as guide for further optimizations.
+
+Ideally, the profile should be generated by a representative workload of the program.  Moreover, the built-in instrumentation impacts the overall performance of the binary.  Both things can make it hard to do profiling of real world applications, because it's not always possible to simulate a representative workload or because of the performance penalty.
```

Comment:
```suggestion
It can be hard to do profiling of real world applications. Ideally, the profile should be generated by a representative workload of the program, but it's not always possible to simulate a representative workload.  Moreover, the built-in instrumentation impacts the overall performance of the binary which introduces a performance penalty.
```